### ISSUE TITLE
Code 

### ISSUE THREAD

**Issue by nomaad42** [ruiz-manuel/yolov7-pose-custom#1]
Hi @ruiz-manuel ! 

Would you mind sharing the training code for custom num of keypoints? 

That would be really awesome!

**Comment by ruiz-manuel** [comment]
I have added the code. Should work for any number of keypoints. Maybe there is only one harcoded list for the colors of the keypoints in plots.py harcoded for 4 keypoints which is what I have. Let me know if you have any issues

**Comment by nomaad42** [comment]
> I have added the code. Should work for any number of keypoints. Maybe there is only one harcoded list for the colors of the keypoints in plots.py harcoded for 4 keypoints which is what I have. Let me know if you have any issues

Thank you very much! 
Let me try to train the model with custom keypoints. 

Btw, how was the performance results of your trained model?

**Comment by ruiz-manuel** [comment]
The keypoints train very well (not sure how to put it in comparable quantitative terms) the classification is good but was a lot better when I trained without keypoints. 0.99 vs .85 mAP. Still poking around with the hyper parameters. I advice you to adjust the loss gains to account for different tasks learning faster than others. I think altering the number of keypoints/classes messes up with that

**Comment by nomaad42** [comment]
Thank you very much for your advice! 

After researching the code, I started the training.
I had several issues, let me ask them: 

1. During training, I have faced "nan" values in "mloss" and "loss_items". It was preventing me to find the "kpt" and "total". 

I added torch command "torch.nan_to_num_(mloss)" line, is that okay?  
I had to add this way at a line 345 in train.py: 

`
            
           if rank in [-1, 0]:


                # print("MLOSS: ", mloss)
                # print("LOSS ITEM: ", loss_items)

                torch.nan_to_num_(mloss)
                torch.nan_to_num(loss_items)

                mloss = (mloss * i + loss_items) / (i + 1)  # update mean losses
                mem = '%.3gG' % (torch.cuda.memory_reserved() / 1E9 if torch.cuda.is_available() else 0)  # (GB)
` 

Why do you think "nan" values happen? Is this exploding gradients problem or some "inf" values in loss tensors?

**Comment by nomaad42** [comment]
2. During training, when I train, after 9 epochs: why performance does not change. I attached the screenshot, why do you think it happens? How can I change that?  
![image](https://user-images.githubusercontent.com/62198792/224301593-efdb00ec-fc85-4cbc-ba39-acb154182432.png)
